client/components/TestQueueRun/index.jsx

```diff
@@ -35,91 +59,192 @@ class TestQueueRow extends Component {
+                    >
+                        Unassign Testers
+                    </Dropdown.Toggle>
+                    <Dropdown.Menu>
+                        {canUnassignTesters.map(t => {
+                            return (
+                                <Dropdown.Item
+                                    as="button"
+                                    key={nextId()}
+                                    value={t.id}
+                                    onClick={this.unassignTesterSelect}
+                                >
+                                    {t.username}
+                                </Dropdown.Item>
+                            );
+                        })}
+                    </Dropdown.Menu>
+                </Dropdown>
+            </Fragment>
+        );
+    }
+
+    renderAssignSelfButton(currentUserAssigned) {
+        const { apgExampleName, atName, browserName } = this.props;
 
         let testrun = `${apgExampleName} for ${atName} on ${browserName}`;
-        let assignOrUnassignButton;
+
         if (currentUserAssigned) {
-            assignOrUnassignButton = (
+            return (
                 <Button
-                    onClick={this.handleUnassignClick}
+                    onClick={this.handleUnassignSelfClick}
                     aria-label={`Unassign me from the test run ${testrun}`}
                 >
                     Unassign Me
                 </Button>
             );
-        } else if (testers.length < 2) {
-            assignOrUnassignButton = (
+        } else {
+            return (
                 <Button
-                    onClick={this.handleAssignClick}
+                    onClick={this.handleAssignSelfClick}
                     aria-label={`Assign me to the test run ${testrun}`}
                 >
                     Assign Me
                 </Button>
             );
         }
+    }
+
+    renderTesterList(currentUserAssigned) {
+        const { testers, userId } = this.props;
 
         let userInfo = testers
             .filter(uid => uid !== userId)
             .map(uid => this.renderTestsCompletedByUser(uid));
-
         if (currentUserAssigned) {
             userInfo.unshift(this.renderTestsCompletedByUser(userId));
         }
 
+        return userInfo;
+    }
+
+    render() {
+        const {
+            admin,
+            cycleId,
+            userId,
+            runId,
+            testers,
+            apgExampleName
+        } = this.props;
+        let currentUserAssigned = testers.includes(userId);
+
+        let designPatternLinkOrName;
+        if (currentUserAssigned) {
+            designPatternLinkOrName = (
+                <Link to={`/cycles/${cycleId}/run/${runId}`}>
+                    {apgExampleName}
+                </Link>
+            );
+        } else {
+            designPatternLinkOrName = apgExampleName;
+        }
+
+        let assignOptions;
+        if (admin) {
```

Comment:
Nonblocking: I like what you did here! Maybe in the long term, we can separate these in the separate components for admin and tester

Reply:
yeah I was also thinking about that, I have more work to do on this component so I might separate them out in the next couple days